### PR TITLE
Update READMEs and make pre-commit work for Frontend

### PR DESCRIPTION
**Pull request by HaGuesto**
As always, I did many steps now that I should not have done when following our Contribution guidelines and I am very sorry for being hypocritical here.

@mc18g13 started here to fix pre-commit and I took it from there and managed last week together with @pylipp to make pre-commit also work for the frontend.

When I started testing, I crashed my set-up and ended up completely re-setting my environment. While walking through all the installation steps, I wrote down all the problems I was faceing and updated / clean-up the READMEs. 

I should not have this much time on this PR and I should have communicated earlier that this is taking me way more time then expected - A great example for feature creep.

Anyway, I would that all of you will profit from this PR and you should *check out the changes to the READMEs* at least. 

After merging other open PRs I will then do a complete run with the linters and formatters and get everything into one format.

### CODE REVIEW

**Comment by HaGuesto** [comment]
[Solve pre-commit config errors #tiny](https://trello.com/c/FpKtPPHH/657-solve-pre-commit-config-errors-tiny)

**Review by billyhao12 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -3,77 +3,75 @@
...
 
-* Install [Docker](https://www.docker.com/products/docker-desktop)
-* Get in touch with the [Boxwise team](mailto:hello@boxwise.co) to get access to the [Auth0](https://auth0.com/) development tenant.
+- Install [Docker](https://www.docker.com/products/docker-desktop)
+- Get in touch with the [Boxwise team](mailto:hello@boxwise.co) to get access to the [Auth0](https://auth0.com/) development tenant.
 
 ## How do I get set up?
 
-1. Create a .env-file with the environment variables. Therefore, copy/paste `example.env` and rename it `.env`
+These are the most basic steps to get frontend and backend up and running. At the end, there are also further instructions to set-up your frontend and backend environment.
```
Comment (aerinsol) [inline README.md:30]: Wait, isn't the first and second statement the same? Are you saying the further instructions will configure more complex features for the environment?

File: flask/README.md [hunk 2]
```diff
@@ -1,24 +1,124 @@
...
+
+## Development Set-up
+
+### Install python
+
+For almost all features of our development set-up you should [install Python >=3.6](https://www.python.org/downloads/) on your computer. You will need it to run tests on the backend and the formatters and linters on both backend and frontend.
+
+**Recommendations for backend people**
+
+These recommendations are mainly ment for people developing on the backend. If you are just a frontend person, but would like use pre-commit and the linters and formatters defined there, you can skip these steps.
```
Comment (aerinsol) [inline flask/README.md:25]: "ment" should be "meant"

File: react/README.md [hunk 3]
```diff
@@ -1,36 +1,50 @@
-# Boxwise react client 
+# Readme
 
-This front-end project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+This frontend project of Boxtribute was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (aerinsol) [inline react/README.md:3]: I think in our brand style guide we agreed that we should use "Front-end" and "back-end" rather than frontend and backend. We should agree and be consistent with this, so let's do a find-and-replace, or make me a good case for why we should update the messaging guidelines.

**Review by aerinsol — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  // only enable if you are using a virtual environment
+  "python.pythonPath": ".venv/bin/python",
```
Comment (pylipp) [inline .vscode/settings.json:3]: I think on Windows the path is `.venv/Scripts/python`. Can anyone confirm?

File: react/README.md [hunk 5]
```diff
@@ -1,36 +1,50 @@
...
+Following the [general set-up steps](../README.md), here a few steps that make your live easier working on the frontend.
 
-The page will reload if you make edits.<br />
-You will also see any lint errors in the console.
+### Set-up pre-commit
 
-### `yarn test`
+[Pre-commit](https://pre-commit.com/) enables us to run code quality checks, such as missing semicolons, trailing whitespace, and debug statements, before you are committing your code. We chose pre-commit since it enables us to run these checks for both frontend and backend in just one place.
+The downside is that you need python to be installed on your computer.
+Please check the [backend README}(../flask/README.md#set-up-pre-commit) to set it up.
```
Comment (pylipp) [inline react/README.md:23]: } should be a ] such that the link is correctly displayed

File: react/README.md [hunk 6]
```diff
@@ -1,36 +1,50 @@
...
+## Table of Contents
 
-In the project directory, you can run:
+1. [Contribution Guidelines](../CONTRIBUTING.md)
+2. [Development Set-up](#development-set-up)
+   1. [Set-up pre-commit](#set-up-pre-commit)
+   2. [Install node and yarn](#install-node-and-yarn)
+   3. [Linting and Formatting in VSCode](#linting-and-formatting-in-vscode)
+3. [Note about yarn and Docker](#note-about-yarn-and-docker)
+4. [Testing][#testing]
```
Comment (pylipp) [inline react/README.md:13]: second [ ] should be ( ) for correct markdown

File: react/README.md [hunk 7]
```diff
@@ -1,36 +1,50 @@
...
+The settings that these extensions are used are already defined in [`.vscode/settings.json`](../.vscode/settings.json).
 
-## Learn More
+## Note about yarn and Docker
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+We are using docker to spin up our dev environment. The react folder is in sync with the react Docker container. Therefore, the hot-reloading of the yarn development server should function.
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+When you wish to add a dependency, e.g. when you make a change to your local package.json, you will need to rebuild the docker container and relaunch.
```
Comment (pylipp) [inline react/README.md:40]: maybe use `package.json`, as below

File: react/README.md [hunk 8]
```diff
@@ -1,36 +1,50 @@
...
+
+You can add packages during development without rebuilding by installing it inside the container. Your changes will last until you "docker-compose down" and will be saved on host for next build.
+
+For example, to add XYZ to the `package.json` file in the `react` folder while developing, you can run this:
+
+      docker-compose exec react yarn add XYZ
+
+(This advice has come from https://github.com/BretFisher/node-docker-good-defaults)
+
+## Testing
```
Comment (pylipp) [inline react/README.md:50]: To do :) ?

File: flask/README.md [hunk 9]
```diff
@@ -1,24 +1,124 @@
...
-This project is developed in Python >= 3.6. For setting up the development environment for the first time, create a [Python virtual environment](https://docs.python.org/3.6/library/venv.html), e.g. by
+## Table of Contents
+1. [Contribution Guidelines](../CONTRIBUTING.md)
+2. [Development Set-up](#development-set-up)
+    1. [Install python](#install-python)
+    2. [Set-up pre-commit](#set-up-pre-commit)
+    3. [Linting and Formatting in VSCode](#linting-and-formatting-in-vscode)
+    4. [Working with MySQL](#working-with-mysql)
+    5. [Debugging](#debugging)
+3. [Testing][#testing]
```
Comment (pylipp) [inline flask/README.md:12]: use ( ) instead of [ ] for correct markdown

File: flask/README.md [hunk 10]
```diff
@@ -1,24 +1,124 @@
...
-Have a look at [direnv](https://github.com/direnv/direnv) if you're interested in ways to automate this procedure. There is already a `.envrc` file in the root of this repo. If you install `direnv` and allow to run it for your local repo, it will access the python virtual environment `venv` everytime you enter the folder via a command line.
+Since we are working with docker you do not have to install a local MySQL server on your computer. Instead, you can just connect to the MySQL server in one of the Docker containers.
 
-### Testing
+#### General notes on Docker network
 
-#### Writing tests
+In the docker-compose file we define a separate docker network called `backend` to which the backend containers are joined. Each container can now look up the hostname `flask` or `mysql` and get back the appropriate container’s IP address.
+To access the mysql database, there are now two possibilities:
+1. You reach the mysql db at `MYSQL_HOST=mysql` and `MYSQL_PORT=3306` or
```
Comment (pylipp) [inline flask/README.md:68]: By this you mean the dev should use the mysql command line client on their host machine?

**Review by pylipp — CHANGES_REQUESTED** [review]
Big changes in the overview! Most things I found were regarding markdown formatting.
Will try to do a functional review by re-setting up the backend env tomorrow :)

File: .vscode/settings.json (hunk 4 above)
Comment (HaGuesto) [reply .vscode/settings.json:3]: how annoying. no clue if that is correct

File: README.md (hunk 1 above)
Comment (HaGuesto) [reply README.md:30]: updated. clarified

File: flask/README.md [hunk 11]
```diff
@@ -1,24 +1,124 @@
...
     python3 -m venv .venv
+
+(optional) 0.2 Activate the virtual environment. If you are not using direnv, be aware that you should do this step each time before working with python in the Boxtribute repo.
+
     source .venv/bin/activate
-    pip install -e . -r requirements-dev.txt
+
+1. Install pre-commit and the linters/formatters (all declared in `/flask/requirements-dev.txt`)
+
+    pip install -e flask -r flask/requirements-dev.txt
```
[the remark this answers is not in the document]
Comment (HaGuesto) [reply flask/README.md:46]: done

File: flask/README.md (hunk 9 above)
Comment (HaGuesto) [reply flask/README.md:12]: done

File: flask/README.md [hunk 12]
```diff
@@ -1,24 +1,124 @@
...
+(optional) 0.1 If you have not already done it, create a venv in which you are running pre-commit. If you are using pyenv you might want to check which python version you are using with `pyenv version` or `which python`.
 
     python3 -m venv .venv
+
+(optional) 0.2 Activate the virtual environment. If you are not using direnv, be aware that you should do this step each time before working with python in the Boxtribute repo.
+
     source .venv/bin/activate
-    pip install -e . -r requirements-dev.txt
+
+1. Install pre-commit and the linters/formatters (all declared in `/flask/requirements-dev.txt`)
```
[the remark this answers is not in the document]
Comment (HaGuesto) [reply flask/README.md:44]: done

File: flask/README.md (hunk 10 above)
Comment (HaGuesto) [reply flask/README.md:68]: I just copy pasted this section from the general README to the flask README. It has nothing to do with wether you want to use the command line client or workbench. Just how to reach the mysql container in your network

File: flask/README.md [hunk 13]
```diff
@@ -1,24 +1,124 @@
...
+
+You can now set break-points in your code.
+If you want to debug a certain endpoint, set a break-point in the endpoint and call this enpoint at the port 5001, e.g.
+        `localhost:5001/api/public`
+If you want to break on any other code lines (not endpoints), then you can only catch them during the server start-up.
+
+#### Usage of Logger
+To log to the console from inside the docker container, create an instance of app using:
+    `from flask import Flask`
+    `app = Flask(__name__)`
```
[the remark this answers is not in the document]
Comment (HaGuesto) [reply flask/README.md:114]: done

File: flask/README.md [hunk 14]
```diff
@@ -28,7 +128,7 @@ Two types of tests can be setup. Model (unit) tests and endpoint (integration) t
 
 New test files should begin with the word test so the they are discovered when running pytest.
 for example:
-```
+. ```
```
[the remark this answers is not in the document]
Comment (HaGuesto) [reply flask/README.md]: done

File: react/README.md (hunk 5 above)
Comment (HaGuesto) [reply react/README.md:23]: done

File: react/README.md (hunk 6 above)
Comment (HaGuesto) [reply react/README.md:13]: done

File: react/README.md (hunk 8 above)
Comment (HaGuesto) [reply react/README.md:50]: yeah for @billyhao12 to fill with wisdom

File: README.md [hunk 15]
```diff
@@ -3,134 +3,111 @@
...
+**Codes connected to existing Boxes in the seed**
+
+![387b0f0f5e62cebcafd48383035a92a](docs/qr/existing/387b0f0f5e62cebcafd48383035a92a.png) ![cba56d486db6d39209dbbf9e45353c4](docs/qr/existing/cba56d486db6d39209dbbf9e45353c4.png) ![a61e0efe25b75032b91106372674c26](docs/qr/existing/a61e0efe25b75032b91106372674c26.png) ![f6f20e805192618def2cb400776a2aa](docs/qr/existing/f6f20e805192618def2cb400776a2aa.png) ![12ca607ce60c484bdbb703def950c5b](docs/qr/existing/12ca607ce60c484bdbb703def950c5b.png)
+![13f12820c8010f2f7349962930e6bf4](docs/qr/existing/13f12820c8010f2f7349962930e6bf4.png) ![d0e144a0a4dc0d8af55e2b686a2e97e](docs/qr/existing/d0e144a0a4dc0d8af55e2b686a2e97e.png) ![69107b2e2b4157b5efe10415bc0bba0](docs/qr/existing/69107b2e2b4157b5efe10415bc0bba0.png) ![b8f0730d36571e4149ba3862379bb88](docs/qr/existing/b8f0730d36571e4149ba3862379bb88.png) ![e1fdfdd942db0e764c9bea06c03ba2b](docs/qr/existing/e1fdfdd942db0e764c9bea06c03ba2b.png)
+
+**Codes not yet connected to Boxes in the seed**
```
Comment (HaGuesto) [inline README.md:85]: @matt-rose6 I added here a few QR-codes for you!

File: flask/README.md [hunk 16]
```diff
@@ -1,112 +1,194 @@
...
+
+### Set-up pre-commit
+
+[Pre-commit](https://pre-commit.com/) enables us to run code quality checks, such as missing semicolons, trailing whitespace, and debug statements, before you are committing your code. We chose pre-commit since it enables us to run these checks for both front-end and back-end in just one place.
+Please follow these steps to set-up pre-commit:
+
+(optional) 0.1 If you have not already done it, create a venv in which you are running pre-commit. If you are using pyenv you might want to check which python version you are using with `pyenv version` or `which python`.
```
Comment (pylipp) [inline flask/README.md:36]: Honestly I would not formulate it as an option, but as a single, very strong recommendation to use a Python venv. I think it's somewhat the basic standard for any Python project. If any dev is more experienced in the Python ecosystem already, they can do as they like. Is anyone in the team working with pyenv?
tldr: mention venv as only option to not confuse new Python devs, or front-end devs who need pre-commit

File: flask/README.md [hunk 17]
```diff
@@ -1,112 +1,194 @@
...
+
+- [Use a version control for python like pyenv.](https://github.com/pyenv/pyenv) It provides you with much more clarity which version you are running and makes it easy to switch versions of python.
+- [Make use of virtual environments like venv.](https://docs.python.org/3/library/venv.html) You should not install all packages you need for this project globally, but focused on Boxtribute.
+- [Have a look at direnv >= v2.21.0](https://github.com/direnv/direnv). Virtual environments must be activated and deactivated. If you are moving through folders in the terminal it can easily happen that you either miss activating or deactivating the venv resulting in errors and time wasted for development. With direnv you can automate the activation and deactivation of venv depending on which folder you are in. There is already a `.envrc` file in the root of this repo. If you install `direnv` and allow to run it for your local repo, it will access the python virtual environment `venv` everytime you enter the folder via a command line.
```
Comment (pylipp) [inline flask/README.md:29]: 'activate' instead of 'access' in the last sentence

File: flask/README.md (hunk 16 above)
Comment (HaGuesto) [reply flask/README.md:36]: yeah would say it is only optional for front-end devs who just want pr-commit. will make it clearer.

File: flask/README.md (hunk 17 above)
Comment (HaGuesto) [reply flask/README.md]: done

**Review by pylipp — APPROVED** [review]
